lib/memo_wise.rb

```diff
@@ -106,21 +108,55 @@ def reset_memo_wise(method_name, *args, **kwargs)
     if args.empty? && kwargs.empty?
       @_memo_wise.delete(method_name)
     else
-      method = self.class.instance_method(method_name)
-
-      has_arg = MemoWise.has_arg?(method)
-
-      if has_arg && MemoWise.has_kwarg?(method)
-        @_memo_wise[method_name].delete([args, kwargs])
-      elsif has_arg
-        @_memo_wise[method_name].delete(args)
-      else
-        @_memo_wise[method_name].delete(kwargs)
-      end
+      @_memo_wise[method_name].delete(fetch_key(method_name, *args, **kwargs))
     end
   end
 
   def reset_all_memo_wise
     @_memo_wise.clear
   end
+
+  def preset_memo_wise(method_name, *args, **kwargs, &block)
+    validate_memo_wised!(method_name)
+
+    unless block
+      raise ArgumentError,
+            "Pass a block as the value to preset for #{method_name}, #{args}"
+    end
+
+    validate_params!(method_name, args)
+
+    if method(method_name).arity.zero?
+      @_memo_wise[method_name] = block_given? ? yield : nil
+    else
+      @_memo_wise[method_name][fetch_key(method_name, *args, **kwargs)] =
+        block_given? ? yield : nil
+    end
+  end
+
+  private
+
+  def validate_memo_wised!(method_name)
+    original_memo_wised_name = :"_memo_wise_original_#{method_name}"
+
+    unless self.class.private_method_defined?(original_memo_wised_name)
+      raise ArgumentError, "#{method_name} is not a memo_wised method"
+    end
+  end
+
+  def fetch_key(method_name, *args, **kwargs)
+    method = self.class.instance_method(method_name)
+    has_arg = MemoWise.has_arg?(method)
+
+    if has_arg && MemoWise.has_kwarg?(method)
+      [args, kwargs]
```

Comment:
In the code above we call `.freeze` on this compound key. Should we add that here or remove that there? (I don't know.)